graylog2-web-interface/src/components/events/events/EventDetails.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import lodash from 'lodash';
+
+import useCurrentUser from 'hooks/useCurrentUser';
+import usePluginEntities from 'hooks/usePluginEntities';
+import { isPermitted } from 'util/PermissionsMixin';
+import { Link } from 'components/common/router';
+import { Col, Row } from 'components/bootstrap';
+import { Timestamp } from 'components/common';
+import Routes from 'routing/Routes';
+import EventDefinitionPriorityEnum from 'logic/alerts/EventDefinitionPriorityEnum';
+import type { Event, EventDefinitionContext } from 'components/events/events/types';
+import ReplaySearchButton from 'views/components/widgets/ReplaySearchButton';
+
+type Props = {
+  event: Event,
+  eventDefinitionContext: EventDefinitionContext,
+};
+
+const EventDetails = ({ event, eventDefinitionContext }: Props) => {
+  const eventDefinitionTypes = usePluginEntities('eventDefinitionTypes');
+
+  const getConditionPlugin = (type: string): EventDefinitionType => {
+    if (type === undefined) {
+      return null;
+    }
+
+    return eventDefinitionTypes.find((edt) => edt.type === type);
+  };
+
+  const renderEventFields = (eventFields: Object[]) => {
```

Comment:
Can we declare this outside of the component? This way it gets redefined on every render, effectively thrashing React's rerendering optimizations. Also, we can probably treat it as a proper React component (like `EventFields` or sth).